pkg/gatherers/clusterconfig/gather_virtual_machine_instances.go

```diff
@@ -0,0 +1,112 @@
+package clusterconfig
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"k8s.io/klog/v2"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"k8s.io/client-go/dynamic"
+
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+// GatherVirtualMachineInstances Collects `VirtualMachineInstance` resources from cluster if available.
+//
+// ### API Reference
+// None
+//
+// ### Sample data
+// - docs/insights-archive-sample/config/virtualmachineinstances/openshift-cnv/fedora-r2nf0eocvxbkmqjy.json
+//
+// ### Location in archive
+// | Version   | Path														|
+// | --------- | ---------------------------------------------------------- |
+// | >= 4.14   | config/virtualmachineinstances/{namespace}/{name}.json		|
+//
+// ### Config ID
+// `clusterconfig/virtual_machine_instances`
+//
+// ### Released version
+// - 4.14
+//
+// ### Backported versions
+// None
+//
+// ### Notes
+// None
+func (g *Gatherer) GatherVirtualMachineInstances(ctx context.Context) ([]record.Record, []error) {
+	gatherDynamicClient, err := dynamic.NewForConfig(g.gatherKubeConfig)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	return gatherVirtualMachineInstances(ctx, gatherDynamicClient)
+}
+
+func gatherVirtualMachineInstances(ctx context.Context, dynamicClient dynamic.Interface) ([]record.Record, []error) {
+	virtualizationList, err := dynamicClient.Resource(virtualMachineInstancesResource).List(ctx, metav1.ListOptions{})
+	if errors.IsNotFound(err) {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	var errs []error
+	// Limit the number of gathered virtualmachineinstances.kubevirt.io
+	var limit = 5
+	records := make([]record.Record, 0, limit)
+	for i := range virtualizationList.Items {
+		item := &virtualizationList.Items[i]
+		records = append(records, record.Record{
+			Name: fmt.Sprintf("config/virtualmachineinstances/%s/%s", item.GetNamespace(), item.GetName()),
+			Item: record.ResourceMarshaller{Resource: anonymizeVirtualMachineInstances(item)},
+		})
+		// limit the gathered records
+		if len(records) == limit {
+			err = fmt.Errorf("limit %d for number of gathered %s resources exceeded", limit, virtualMachineInstancesResource.GroupResource())
+			errs = append(errs, err)
+			break
+		}
+	}
+
+	return records, errs
+}
+
+func anonymizeVirtualMachineInstances(data *unstructured.Unstructured) *unstructured.Unstructured {
+	const errMsg = "error during anonymizing virtualmachineinstances:"
+	volumes, found, err := unstructured.NestedSlice(data.Object, "spec", "volumes")
```

Comment:
you can check and use https://github.com/openshift/insights-operator/blob/master/pkg/utils/unstructerd_wrappers.go#L22

Reply:
Nice! I wasn't aware of this helper. I'll update the code. Thanks.